utils.go

```diff
@@ -16,7 +16,15 @@ func newHTTPClient(u *url.URL, tlsConfig *tls.Config, timeout time.Duration) *ht
 	switch u.Scheme {
 	default:
 		httpTransport.Dial = func(proto, addr string) (net.Conn, error) {
-			return net.DialTimeout(proto, addr, timeout)
+			conn, err := net.DialTimeout(proto, addr, timeout)
+			if tcpConn, ok := conn.(*net.TCPConn); ok {
+				// Set TCP user timeout. Sender breaks TCP connection
+				// if packets are not acknowledged after 20 seconds. This is a
+				// relatively new TCP option to improve dead peer detection.
+				// Do not fail newHTTPClient if OS doesn's support it.
+				SetTCPUserTimeout(tcpConn, 20*time.Second)
```

Comment:
I'm not very familiar with this, but I'm wondering, why hard code 20s and not use `timeout` ?